fix: fix varint decoding

stringhandler opened the pull request.

Commit:
fix: do not allow ambigious varint encodings

Commit:
fix: check for overflow during deserialize

Commit:
test: add fuzzing

Commit:
chore: add idea files

hansieodendaal reviewed (approved):
Please run `cargo fmt --all`
I think we need some CI stuff as well

h4sh3d committed:
Chore: update deps & ci (#175)

dependabot[bot] committed:
chore(deps): bump actions/checkout from 3 to 4 (#171)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump Swatinem/rust-cache from 2.6.0 to 2.7.0 (#173)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: h4sh3d <h4sh3d@protonmail.com>

h4sh3d committed:
Doc: improve serde amount modules (#176)

h4sh3d committed:
chore: update changelog (#178)

github-actions[bot] committed:
Release version 0.19.0 (#177)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Commit:
remove fuzz target and fmt

Commit:
cleanup cargo.toml

stringhandler committed:
Merge branch 'main' into st-advisory-fix2

Boog900 committed:
fix: transaction scanning (#180)

Change tx scanning to use both the main public key and the additional key at that outputs idx

Commit:
remove idea files

Commit:
Merge remote-tracking branch 'mike/st-advisory-fix2' into st-advisory-fix2

Commit:
remove idea files

Commit:
Merge remote-tracking branch 'origin/main' into st-advisory-fix2

Commit:
fix change log

stringhandler commented:
Closing as I'm not following up on this anymore
